### DISCUSSION TITLE
Serialization of shadow properties

### DISCUSSION THREAD

**Discussion by kalu06** [mikro-orm/mikro-orm#5295] [General]
While playing with the serialization, I observed that no matter what `fields` contains (when using in `.find()` for example), the shadow properties (not marked as hidden) are always returned in the serialized object.

But when hiding them (`{ ..., hidden: true }`), there's no way to have those in the serialized object.

Is there a way to have more control and flexibility over shadow property serialization?

Thanks

**Comment by B4nan** [comment]
> But when hiding them ({ ..., hidden: true }), there's no way to have those in the serialized object.

But that's consistent with the rest, isn't it? If you mark things as hidden, they will never be part of the serialized object, regardless of the `fields` hint, right? Because `hidden` is a serialization flag, while `fields` is about the actual query and what it returns.

> Is there a way to have more control and flexibility over shadow property serialization?

Have you tried explicit serialization? If you add the property to the `populate` hint there, it should be visible regardless of the `hidden` option, as that always takes precedence with explicit serilalization.

**Reply by kalu06** [reply]
Yes, indeed it makes sense. The proper solution is to indeed use explicit serialization with `serialize`.
Thanks again for your prompt reply Martin.
